e2e/src/auction_bid.rs

```diff
@@ -0,0 +1,87 @@
+use ethcontract::web3::types::{H160, H256, U256};
+
+#[derive(Debug)]
+pub struct AuctionBid {
+    order_hash: H256,
+    num_orders: u64,
+    creation_time_stamp: U256,
+    pub solver: H160,
+    objective_value: U256,
+    pub tentative_state: H256,
+    solution_hash: H256,
+}
+
+type PreAuctionBid = (
+    [u8; 32],
+    u64,
+    U256,
+    H160,
+    U256,
+    [u8; 32],
+    [u8; 32],
+    U256,
+    U256,
+);
+
+impl AuctionBid {
+    pub fn from(contract_bid: &PreAuctionBid) -> Self {
+        let order_hash = H256::from_slice(&contract_bid.0);
+        let num_orders = contract_bid.1;
+        let creation_time_stamp = contract_bid.2;
+        let solver = contract_bid.3;
+        let objective_value = contract_bid.4;
+        let tentative_state = H256::from_slice(&contract_bid.5);
+        let solution_hash = H256::from_slice(&contract_bid.6);
+        let _auction_applied_time = contract_bid.7;
+        let _applied_account_state_index = contract_bid.8;
+        AuctionBid {
+            order_hash,
+            num_orders,
+            creation_time_stamp,
+            solver,
+            objective_value,
+            tentative_state,
+            solution_hash,
+        }
+    }
+}
+
+#[cfg(test)]
+pub mod tests {
+    use super::*;
+    use std::str::FromStr;
+
+    #[test]
+    fn test_auction_bid_new() {
```

Comment:
It's a bit weird to test your test helper structs. Usually a sign that the tests are over-engineered.

Reply:
Yes, I agree and was considering simply removing the AuctionBid model completely since I only really use it to abstract away the parsing of a smart contract read. For now I will remove the test and  use the other helper function you suggested to clean thing up a bit.